sdk/azidentity/aad_identity_client.go

```diff
@@ -74,14 +74,14 @@ func (c *aadIdentityClient) authenticate(ctx context.Context, tenantID string, c
 
 	// This should never happen under normal conditions
 	if resp == nil {
-		return nil, &AuthenticationFailedError{Message: "Something unexpected happened with the request and received a nil response"}
+		return nil, &AuthenticationFailedError{inner: errors.New("Something unexpected happened with the request and received a nil response")}
```

Comment:
I think the `msg` field should be set to this.  `inner` should only be used if it's wrapping an error returned from some other operation.

Reply:
Totally agree.